### PR TITLE
chore(web): improve select and unselect

### PR DESCRIPTION
**Pull request by mkumbobeaty**
# Overview

-  Improve select and unselect:
   - layers
   - scene setting
   - layer styles

### CODE REVIEW

File: web/src/beta/features/Editor/useLayerStyles.ts [hunk 1]
```diff
@@ -37,8 +37,7 @@ export default function ({ sceneId }: LayerStyleProps) {
   );
 
   const handleLayerStyleSelect = useCallback(
-    (layerId: string) =>
-      setSelectedLayerStyleId(prevId => (prevId === layerId ? undefined : layerId)),
+    (layerId: string) => setSelectedLayerStyleId(layerId ? layerId : undefined),
```
Comment (KaWaite) [inline web/src/beta/features/Editor/useLayerStyles.ts:40]: layerId will never be undefined
```suggestion
    (layerId: string) => setSelectedLayerStyleId(layerId),
```

File: web/src/beta/features/Editor/useLayers.ts [hunk 2]
```diff
@@ -52,17 +52,13 @@ export default function ({ sceneId, isVisualizerReady, visualizerRef }: LayerPro
   const handleLayerSelect = useCallback(
     (layerId?: string) => {
       if (!isVisualizerReady) return;
+      layerId ? setSelectedLayerId({ layerId }) : setSelectedLayerId(undefined);
```
Comment (KaWaite) [inline web/src/beta/features/Editor/useLayers.ts:55]:
```suggestion
setSelectedLayerId(layerId ? { layerId } : undefined);
```

**Review by KaWaite — APPROVED** [review]

File: web/src/beta/features/Editor/useScene.ts [hunk 3]
```diff
@@ -18,11 +18,8 @@ export default function ({ sceneId }: SceneProps) {
     [scene?.property, selectedSceneSetting],
   );
   const handleSceneSettingSelect = useCallback(
-    (collection?: string) =>
-      setSelectedSceneSetting(
-        !collection || selectedSceneSetting === collection ? undefined : collection,
-      ),
-    [selectedSceneSetting, setSelectedSceneSetting],
+    (collection?: string) => setSelectedSceneSetting(!collection ? undefined : collection),
```
Comment (airslice) [inline web/src/beta/features/Editor/useScene.ts:21]:
```suggestion
    (collection?: string) => setSelectedSceneSetting(collection),
```

File: web/src/beta/features/Editor/useLayers.ts (hunk 2 above)
Comment (airslice) [reply web/src/beta/features/Editor/useLayers.ts:55]: I'm not very sure about the detail of select states between editor & core right now. 
But i think it's good to make the logic on editor part more straight forward & make sure there's no obvious bugs for now, and maybe we can have a deep look into the selections and do some refactors later together with core when members are free from view3.0 project.

**Comment by airslice** [comment]
Thank you for the work! 
Since this branch is kind of out-of-date and currently we have a deeper refactor on selection system i'll close this PR.

**Pull request closed without merging**